src/Core/Resolvers/CosmosQueryEngine.cs

```diff
@@ -396,7 +400,7 @@ private static async Task<(string? idValue, string? partitionKeyValue)> GetIdAnd
         /// </summary>
         /// <param name="parameter"></param>
         /// <returns></returns>
-        private static string? GetIdValue(object? parameter)
+        private static string? GetIdValue(IMiddlewareContext context, object? parameter)
```

Comment:
Any advantage of passing the entire `context` instead of just `context.Variables`?

Reply:
yes it we are using `selection` from context also https://github.com/Azure/data-api-builder/blob/25012aa14fd2c72e9239b179d6d4f4acda2e4dfb/src/Core/Resolvers/CosmosQueryEngine.cs#L421